libs/ui-lib/lib/ocm/components/clusterConfiguration/manifestsConfiguration/components/customManifestsValidationSchema.tsx

```diff
@@ -7,51 +7,87 @@ import {
   validateFileType,
   INCORRECT_TYPE_FILE_MESSAGE,
 } from '../../../../../common/utils';
-const INCORRECT_FILENAME =
-  'Must have a yaml, yml, json, yaml.patch or yml.patch extension and can not contain /.';
 
-const UNIQUE_FOLDER_FILENAME = 'Ensure unique file names to avoid conflicts and errors.';
+import * as yup from 'yup';
 
-export const getUniqueValidationSchema = Yup.string().test(
+declare module 'yup' {
+  interface ArraySchema<
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    TIn extends any[] | null | undefined,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    TContext,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    TDefault = undefined,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    TFlags extends Yup.Flags = '',
+  > {
+    unique(message: string, mapper?: (a: any) => any): this;
+  }
+}
+
+Yup.addMethod(
+  Yup.array,
   'unique',
```

Comment:
This name seems to imply it's a generic "isUnique" check, but it's specific to customManifests (because of the errors it's generating).

So I think it should be renamed, and I would change it so there's no need for so many `// eslint-disable-next-line`:

```
/* eslint-disable */
type CustomManifestMapper = (a: CustomManifestValues) => string;

declare module 'yup' {
  interface ArraySchema<
    TIn extends any[] | null | undefined,
    TContext,
    TDefault = undefined,
    TFlags extends Yup.Flags = '',
  > {
    uniqueManifestFiles(message: string, mapper: CustomManifestMapper): this;
  }
}
/* eslint-enable */
```

Then you can define the function with the proper type signature

```
Yup.addMethod(
  Yup.array,
  'uniqueManifestFiles',
  function (message: string, mapper: CustomManifestMapper) {
    return this.test('unique', message, function (list: CustomManifestValues[] | undefined) {
```

and there is no need for the exception at 
```
const mappedValue = mapper(item);
```